receipt/src/main/java/com/hyperwallet/android/receipt/view/ListReceiptFragment.java

```diff
@@ -0,0 +1,307 @@
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.list_receipt_fragment, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        mProgressBar = view.findViewById(R.id.list_receipt_progress_bar);
+        mListReceiptAdapter = new ListReceiptAdapter(new ListReceiptItemDiffCallback());
+        mListReceiptsView = view.findViewById(R.id.list_receipts);
+        mListReceiptsView.setHasFixedSize(true);
+        mListReceiptsView.setLayoutManager(new LinearLayoutManager(getActivity()));
+        mListReceiptsView.setAdapter(mListReceiptAdapter);
+        registerObservers();
+    }
+
+    private void registerObservers() {
+        mListReceiptViewModel.getReceiptList().observe(getViewLifecycleOwner(), new Observer<PagedList<Receipt>>() {
+            @Override
+            public void onChanged(PagedList<Receipt> transferMethods) {
+                mListReceiptAdapter.submitList(transferMethods);
+            }
+        });
+
+        mListReceiptViewModel.getReceiptErrors().observe(getViewLifecycleOwner(),
+                new Observer<Event<List<HyperwalletError>>>() {
+            @Override
+            public void onChanged(Event<List<HyperwalletError>> listEvent) {
+                if (!listEvent.isContentConsumed()) {
+                    mOnLoadReceiptErrorCallback.showErrorOnLoadReceipt(listEvent.getContent());
+                }
+            }
+        });
+
+        mListReceiptViewModel.isLoadingData().observe(getViewLifecycleOwner(), new Observer<Boolean>() {
+            @Override
+            public void onChanged(Boolean loading) {
+                if (loading) {
+                    mProgressBar.setVisibility(View.VISIBLE);
+                } else {
+                    mProgressBar.setVisibility(View.GONE);
+                }
+            }
+        });
+    }
+
+    void retry() {
+        mListReceiptViewModel.retryLoadReceipts();
+    }
+
+    interface OnLoadReceiptErrorCallback {
+
+        void showErrorOnLoadReceipt(@NonNull final List<HyperwalletError> errors);
+    }
+
+    private static class ListReceiptItemDiffCallback extends DiffUtil.ItemCallback<Receipt> {
+
+        @Override
+        public boolean areItemsTheSame(@NonNull Receipt oldItem, @NonNull Receipt newItem) {
+            return oldItem.hashCode() == newItem.hashCode()
+                    && Objects.equals(oldItem, newItem);
+        }
+
+        @Override
+        public boolean areContentsTheSame(@NonNull Receipt oldItem, @NonNull Receipt newItem) {
+            return oldItem.hashCode() == newItem.hashCode()
+                    && Objects.equals(oldItem, newItem);
+        }
+    }
+
+    private static class ListReceiptAdapter
```

Comment:
probably it would be better to split adapter in a separate class.

Reply:
@vshcherbyna-epam what do you mean by this? can you expand your wisdom about your comment?

Reply:
I was just looking onto moving the Adapter to separate file and keep Fragment a little bit neat.